core/bug_api.php

```diff
@@ -985,10 +985,18 @@ function bug_text_clear_cache( $p_bug_id = null ) {
  * @access public
  */
 function bug_exists( $p_bug_id ) {
-	if( false == bug_cache_row( $p_bug_id, false ) ) {
+	$c_bug_id = (int)$p_bug_id;
+
+	# Prevent db query errors
+	if( $c_bug_id > db_max_uint() ) {
```

Comment:
Why didn't you implement the logic in bug_cache_row? (near to the database code)